soc/arm/ti_simplelink/cc13x2_cc26x2/power.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2019 Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <init.h>
+#include <power/power.h>
+
+#include <ti/drivers/Power.h>
+#include <ti/drivers/power/PowerCC26X2.h>
+
+#include <ti/drivers/dpl/ClockP.h>
+
+#include <ti/devices/cc13x2_cc26x2/driverlib/cpu.h>
+#include <ti/devices/cc13x2_cc26x2/driverlib/vims.h>
+#include <ti/devices/cc13x2_cc26x2/driverlib/sys_ctrl.h>
+
+#include <logging/log.h>
+#define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
+LOG_MODULE_REGISTER(soc);
+
+/* Configuring TI Power module to not use its policy function (we use Zephyr's
+ * instead), and disable oscillator calibration functionality for now.
+ */
+const PowerCC26X2_Config PowerCC26X2_config = {
+	.policyInitFxn      = NULL,
+	.policyFxn          = NULL,
+	.calibrateFxn       = NULL,
+	.enablePolicy       = false,
+	.calibrateRCOSC_LF  = false,
+	.calibrateRCOSC_HF  = false
+};
+
+extern PowerCC26X2_ModuleState PowerCC26X2_module;
+
+/*
+ * Power state mapping:
+ * SYS_POWER_STATE_SLEEP_1: Idle
+ * SYS_POWER_STATE_SLEEP_2: Standby
+ */
+
+/* Invoke Low Power/System Off specific Tasks */
+void sys_set_power_state(enum power_states state)
+{
+#ifdef CONFIG_SYS_POWER_SLEEP_STATES
+#ifdef CONFIG_HAS_SYS_POWER_STATE_SLEEP_1
```

Comment:
When would `CONFIG_HAS_SYS_POWER_STATE_SLEEP_1` not be defined?

Reply:
Good point. I had picked up this construct by looking at other platforms, e.g.

soc/arm/silabs_exx32/common/soc_power.c
soc/arm/nordic_nrf/nrf51/power.c

My guess is that it is a provision just in case a device from the same family is added later that for some reason does not support one of the sleep or deep sleep states. I did the same for consistency, but I think you are right the check is unnecessary on this platform.

Let me know if you feel strongly against having this, in which case I'll go ahead and remove it.